java/wavefrontsender.md

```diff
@@ -0,0 +1,212 @@
+
+* Option 1: [Create a `WavefrontProxyClient`](#Option-1-Sending-Data-via-the-Wavefront-Proxy) to send data to a Wavefront proxy.
+* Option 2: [Create a `WavefrontDirectIngestionClient`](#Option-2-Sending-Data-via-Direct-ingestion) to send data directly to a Wavefront service.
+
+## Option 1: Sending Data via the Wavefront Proxy
+
+>**Prerequisite**
+>Before your application can use a `WavefrontProxyClient`, you must [set up and start a Wavefront proxy](https://github.com/wavefrontHQ/java/tree/master/proxy#set-up-a-wavefront-proxy).
+
+When sending data via the Wavefront proxy, you need to create a `WavefrontProxyClient`. You need to build it with the information needed to send data to a Wavefront proxy, including:
+
+* The name of the host that will run the Wavefront proxy.
+* One or more proxy listening ports to send data to. The ports you specify depend on the kinds of data you want to send (metrics, histograms, and/or trace data). You must specify at least one listener port. 
+* Optional settings for tuning communication with the proxy.
+
+
+```java
+// Create the builder with the proxy hostname or address
+WavefrontProxyClient.Builder wfProxyClientBuilder = new WavefrontProxyClient.Builder(proxyHostName);
+
+// Set the proxy port to send metrics to. Default: 2878
+wfProxyClientBuilder.metricsPort(2878);
+
+// Set a proxy port to send histograms to.  Recommended: 2878
+wfProxyClientBuilder.distributionPort(2878);
+
+// Set a proxy port to send trace data to. Recommended: 30000
+wfProxyClientBuilder.tracingPort(30_000);
+
+// Optional: Set a custom socketFactory to override the default SocketFactory
+wfProxyClientBuilder.socketFactory(<SocketFactory>);
+
+// Optional: Set a nondefault interval (in seconds) for flushing data from the sender to the proxy. Default: 5 seconds
+wfProxyClientBuilder.flushIntervalSeconds(2);
+
+// Create the WavefrontProxyClient
+WavefrontSender wavefrontSender = wfProxyClientBuilder.build();
+ ```
+**Note:** When you [set up a Wavefront proxy](https://github.com/wavefrontHQ/java/tree/master/proxy#set-up-a-wavefront-proxy) on the specified proxy host, you specify the port it will listen to for each type of data to be sent. The `WavefrontProxyClient` must send data to the same ports that the Wavefront proxy listens to. Consequently, the port-related builder methods must specify the same port numbers as the corresponding proxy configuration properties: 
+
+| `WavefrontProxyClient` builder method | Corresponding property in `wavefront.conf` |
+| ----- | -------- |
+| `metricsPort()` | `pushListenerPorts=` |
+| `distributionPort()` | `histogramDistListenerPorts=` |
+| `tracingPort()` | `traceListenerPorts=` |
+ 
+
+## Option 2: Sending Data via Direct ingestion
+
+when sending data via direct ingestion, you need to create a `WavefrontDirectIngestionClient`, and build it with the information it needs to send data directly to Wavefront.
```

Comment:
When (not when)

and build it with the information it needs to send data directly to Wavefront. => What is that information?  the URL and the API token?